internal/services/controller/controller.go

```diff
@@ -236,6 +241,29 @@ func (c *Controller) Run(ctx context.Context) error {
 		c.queue.ShutDown()
 	}()
 
+	podMetricsType := reflect.TypeOf(&v1beta1.PodMetrics{})
+	if c.informers[podMetricsType] == nil {
+		go func() {
+			if err := wait.PollInfiniteWithContext(ctx, time.Minute*2, func(ctx context.Context) (done bool, err error) {
+				if !isMetricsServerAPIAvailable(c.log, c.discovery) {
+					return false, nil
+				}
+
+				c.log.Infof("Cluster supports pod metrics, will start collecting them.")
+				metricsInformer := c.f.InformerFor(&v1beta1.PodMetrics{}, func(_ kubernetes.Interface, _ time.Duration) cache.SharedIndexInformer {
+					return custominformers.NewPodMetricsInformer(c.log, c.metricsClient)
+				})
+				informer := custominformers.NewHandledInformer(c.log, c.queue, metricsInformer, podMetricsType)
```

Comment:
duplicates logic in the "new" - maybe just remove from constructor, and leave only this delayed initialization?

Reply:
Removed the logic completely, will retest